python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -0,0 +1,124 @@
+from typing import List, Iterator, Any, Dict, Callable, Optional
+
+import ray
+from ray.data.block import Block, BlockMetadata
+from ray.data._internal.stats import StatsDict
+from ray.data._internal.compute import (
+    ComputeStrategy,
+    TaskPoolStrategy,
+    ActorPoolStrategy,
+)
+from ray.data._internal.execution.interfaces import (
+    RefBundle,
+    PhysicalOperator,
+)
+from ray.data._internal.execution.operators.map_operator_tasks_impl import (
+    MapOperatorTasksImpl,
+)
+from ray.data._internal.execution.operators.map_operator_actors_impl import (
+    MapOperatorActorsImpl,
+)
+
+
+class MapOperator(PhysicalOperator):
+    """A streaming operator that maps input bundles 1:1 to output bundles.
+
+    This operator implements the distributed map operation, supporting both task
+    and actor compute strategies.
+    """
+
+    def __init__(
+        self,
+        transform_fn: Callable[[Iterator[Block]], Iterator[Block]],
+        input_op: PhysicalOperator,
+        name: str = "Map",
+        compute_strategy: Optional[ComputeStrategy] = None,
+        min_rows_per_batch: Optional[int] = None,
+        ray_remote_args: Optional[Dict[str, Any]] = None,
+    ):
+        """Create a MapOperator.
+
+        Args:
+            transform_fn: The function to apply to each ref bundle input.
+            input_op: Operator generating input data for this op.
+            name: The name of this operator.
+            compute_strategy: Customize the compute strategy for this op.
+            min_rows_per_batch: The number of rows to gather per batch passed to the
```

Comment:
Should we name it as `min_rows_per_fn_call`? `batch` is kind of confusing here, as this is neither user-facing batch in `map_batches`, nor zero-copy batch execution we shall introduce later.

Reply:
Batch seems clearer to me: it basically is the same as the user facing batch size.

Reply:
I don't think so, the previous naming was very confusing for me. The new one is clear in intent.

Reply:
@ericl But that intent is incorrect: this is a target to get near to, not a minimum/floor. We add blocks to a bundle up to this target size, but we purposefully do not exceed it, so this is definitely not a minimum.

Reply:
Alright, let me rename this to `min_rows_per_bundle` then. I don't think it's possible to be precisely unambiguous, and would prefer we keep the "min" intent which is the big picture.